skill-tree /src/main/java/com/RDS/skilltree/utils/FetchAPI.java

```diff
@@ -0,0 +1,33 @@
+package com.RDS.skilltree.utils;
+
+import com.RDS.skilltree.utils.RDSUser.Response;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.concurrent.CompletableFuture;
+
+@Component
+public class FetchAPI {
+    private final RestTemplate restTemplate;
+
+    public FetchAPI(RestTemplateBuilder restTemplateBuilder){
+        restTemplate = restTemplateBuilder.build();
+    }
+
+    @Async
+    public CompletableFuture<Response> getRDSUserData(String userId) {
+        String url = String.format("https://api.realdevsquad.com/users/userId/%s", userId);
+        try{
+            ResponseEntity<Response> response = restTemplate.getForEntity(url, Response.class);
+            Response result = response.getBody();
+            return CompletableFuture.completedFuture(result);
+        }catch (Exception e){
```

Comment:
As user is interacting with skill tree backend, if there is an error in calling any 3rd party service (RDS backend), I think from user POV its internal server error right?
@iamitprakash

Reply:
Here the socket exception, might not happen timeout can happen, try adding the timeout to the website backend call, to avoid waiting for the backend for long times